File: palindrome-number/python/palindrome-number.py
#!/usr/bin/python3

import math
def isPalindrome(x: int) -> bool:
    if x < 0: return False
    r = []
    for i in range(int(math.log10(x)), -1, -1):
        exp = 10**(i)
        f = int(x/exp)
        r.append(f)
        x = x - f*exp
    return r == r[::-1]


if __name__ == "__main__":
    print("Running tests...")
    assert isPalindrome(141) == True
    assert isPalindrome(-141) == False
    assert isPalindrome(10) == False
    print("Passed all tests!")

File: palindrome-number/c/palindrome-number.c
#include <assert.h>
#include <math.h>
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>

bool isPalindrome(int x) {
  long int rev = 0;
  int temp = x;
  if (x < 0)
    return false;

  while (x) {
    rev = rev * 10 + x % 10;
    x /= 10;
  }

  return (temp == rev);
}

int main() {

  printf("Running tests...\n");
  assert(isPalindrome(141) == true);
  assert(isPalindrome(-141) == false);
  assert(isPalindrome(10) == false);
  assert(isPalindrome(1799999971) == true);
  printf("Passed all tests!\n");

  return 0;
}
